File: grid.cpp
#include "grid.h"
#include <iostream>
using std::cout, std::endl;


Grid::Grid(int width, int height, vector<sf::Color> colors, int windowSize){
    unitMatrix = {};
    buildMatrix(width, height, colors, windowSize);
}

void Grid::buildMatrix(int width, int height, vector<sf::Color> colors, int windowSize){
    vector<Unit> row;
    float unitSize = windowSize / width;
    for(int wCounter=0; wCounter<width; wCounter++){
        row = vector<Unit> {};
        for(int hCounter=0; hCounter<height; hCounter++){
            row.push_back(Unit(0, wCounter, hCounter, colors, unitSize));
        }
        unitMatrix.push_back(row);
    }
}

int Grid::getHeight(){
    return unitMatrix.size();
}

int Grid::getWidth(){
    return unitMatrix[0].size();
}


//Below is a testable main function that can be used to test just the grid class.

/*int main(){
    Unit test(0, 0, 0);
    Grid testGrid(10, 10);
    cout << test.getLocation()[0] << endl;
    return 0;
}*/

File: grid.py
import unit

class Grid:

    def __init__(self, width, height):
        self.width = width
        self.height = height
        self.grid = self.build_grid()
        print(self.grid)

    def build_grid(self):
        grid = []
        for x in range(self.width):
            row = []
            for y in range(self.height):
                row.append(unit.Unit("brown"))
            grid.append(row)
        return grid  

    def get_width(self):
        return self.width
    
    def set_width(self, new_width):
        self.width = new_width

    def get_height(self):
        return self.height
    
    def set_height(self, new_height):
        self.height = new_height

def main():
    grid = Grid(150, 150)
    print(grid.get_height())

if __name__ == "__main__":
    main()